src/geode/mesh/helpers/convert_solid_mesh.cpp

```diff
@@ -161,8 +162,168 @@ namespace
         }
         return true;
     }
+
+    std::array< geode::index_t, 8 > order_hexahedron_vertices(
+        const geode::index_t hexahedron_id, const geode::SolidMesh3D& solid )
+    {
+        std::array< geode::index_t, 8 > ordered_vertices;
+        const auto first_polyhedron_facet_vertices =
+            solid.polyhedron_facet_vertices( { hexahedron_id, 0 } );
+        ordered_vertices[0] = first_polyhedron_facet_vertices[0];
+        ordered_vertices[1] = first_polyhedron_facet_vertices[3];
+        ordered_vertices[2] = first_polyhedron_facet_vertices[2];
+        ordered_vertices[3] = first_polyhedron_facet_vertices[1];
+        geode::index_t already_used_facet{ 0 };
+        for( const auto f :
+            geode::Range( 1, solid.nb_polyhedron_facets( hexahedron_id ) ) )
+        {
+            const auto polyhedron_facet_vertices =
+                solid.polyhedron_facet_vertices( { hexahedron_id, f } );
+            if( absl::c_find( polyhedron_facet_vertices, ordered_vertices[0] )
+                    != polyhedron_facet_vertices.end()
+                && absl::c_find(
+                       polyhedron_facet_vertices, ordered_vertices[1] )
+                       != polyhedron_facet_vertices.end() )
+            {
+                for( const auto v_id :
+                    geode::Indices( polyhedron_facet_vertices ) )
+                {
+                    if( polyhedron_facet_vertices.at( v_id )
+                        == ordered_vertices[0] )
+                    {
+                        ordered_vertices[4] =
+                            polyhedron_facet_vertices[( v_id + 1 ) % 4];
+                        ordered_vertices[5] =
+                            polyhedron_facet_vertices[( v_id + 2 ) % 4];
+                        already_used_facet = f;
+                        break;
+                    }
+                }
+                break;
+            }
+        }
+        for( const auto f :
+            geode::Range( 1, solid.nb_polyhedron_facets( hexahedron_id ) ) )
+        {
+            const auto polyhedron_facet_vertices =
+                solid.polyhedron_facet_vertices( { hexahedron_id, f } );
+            if( f == already_used_facet )
+            {
+                continue;
+            }
```

Comment:
I think you could do a single loop and simply check both condition (use continue; to skip second check if first one is true), by seraching for ordered_vertices[2] and [3] in the second loop
To simplify the if/for loop, you can also use the result of the first c_find (it will give you the position of the vertex you search in the array)